dds/DCPS/TypeSupportImpl.cpp

```diff
@@ -49,35 +49,46 @@ TypeSupportImpl::get_type_name()
   return type._retn();
 }
 
-void TypeSupportImpl::to_type_info(XTypes::TypeInformation& type_info) const
+void TypeSupportImpl::to_type_info_i(XTypes::TypeIdentifierWithDependencies& ti_with_deps,
+                                     const XTypes::TypeIdentifier& ti,
+                                     const XTypes::TypeMap& type_map) const
 {
-  using namespace XTypes;
-  const TypeIdentifier& minTypeId = getMinimalTypeIdentifier();
-  const TypeMap& minTypeMap = getMinimalTypeMap();
-  const TypeMap::const_iterator pos = minTypeMap.find(minTypeId);
-
-  if (pos == minTypeMap.end()) {
-    type_info.minimal.typeid_with_size.type_id = TypeIdentifier();
-    type_info.minimal.typeid_with_size.typeobject_serialized_size = 0;
-
+  const XTypes::TypeMap::const_iterator pos = type_map.find(ti);
+
+  if (pos == type_map.end()) {
+    const char* ek = ti.kind() == XTypes::EK_MINIMAL ? "minimal" : "complete";
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: TypeSupportImpl::to_type_info_i, ")
+               ACE_TEXT("%C TypeIdentifier of topic type not found in local type map.\n"), ek));
+    ti_with_deps.typeid_with_size.type_id = XTypes::TypeIdentifier();
+    ti_with_deps.typeid_with_size.typeobject_serialized_size = 0;
   } else if (TheServiceParticipant->type_object_encoding() == Service_Participant::Encoding_WriteOldFormat) {
-    Encoding encoding = get_typeobject_encoding();
+    Encoding encoding = XTypes::get_typeobject_encoding();
     encoding.skip_sequence_dheader(true);
-    const TypeObject& minTypeObject = pos->second;
-    type_info.minimal.typeid_with_size.type_id = makeTypeIdentifier(minTypeObject, &encoding);
-    const size_t sz = serialized_size(encoding, minTypeObject);
-    type_info.minimal.typeid_with_size.typeobject_serialized_size = static_cast<unsigned>(sz);
-
+    const XTypes::TypeObject& to = pos->second;
+    ti_with_deps.typeid_with_size.type_id = makeTypeIdentifier(to, &encoding);
+    const size_t sz = serialized_size(encoding, to);
+    ti_with_deps.typeid_with_size.typeobject_serialized_size = static_cast<unsigned>(sz);
   } else {
-    const TypeObject& minTypeObject = pos->second;
-    type_info.minimal.typeid_with_size.type_id = minTypeId;
-    const size_t sz = serialized_size(get_typeobject_encoding(), minTypeObject);
-    type_info.minimal.typeid_with_size.typeobject_serialized_size = static_cast<unsigned>(sz);
+    ti_with_deps.typeid_with_size.type_id = ti;
+    const XTypes::TypeObject& to = pos->second;
+    const size_t sz = serialized_size(XTypes::get_typeobject_encoding(), to);
+    ti_with_deps.typeid_with_size.typeobject_serialized_size = static_cast<unsigned>(sz);
   }
 
-  type_info.minimal.dependent_typeid_count = 0;
-  type_info.complete.typeid_with_size.typeobject_serialized_size = 0;
-  type_info.complete.dependent_typeid_count = 0;
+  ti_with_deps.dependent_typeid_count = -1;
```

Comment:
Since this `-1` is a special value defined by the spec, we should use a named constant for it.